baselines/jft/active_learning.py

```diff
@@ -659,65 +680,70 @@ def main(config, output_dir, acquisition_method):
   rng, rng_loop = jax.random.split(rng)
   rngs_loop = flax_utils.replicate(rng_loop)
 
-  # NOTE: it's VITAL train_ds_rng is used for all train_ds creations
+  # TODO(joost,andreas): double check if below is still necessary (train_split is independent of this)
+  # NOTE: train_ds_rng is re-used for all train_ds creations
   rng, train_ds_rng = jax.random.split(rng)
 
-  current_train_ds_length = len(
-      train_subset_data_builder.subset_ids[config.train_split])
-  while current_train_ds_length < config.get("max_labels", 250):
+  while True:
+    current_train_ds_length = len(
+        train_subset_data_builder.subset_ids[config.train_split])
+    if current_train_ds_length >= config.get("max_training_set_size", 150):
+      break
     print(f"Training set size: {current_train_ds_length}")
 
     current_opt_repl = flax_utils.replicate(opt_cpu)
 
-    # Repeat dataset to have oversampled epochs and bootstrap more batches
-    number_of_batches = current_train_ds_length / config.batch_size
-    num_repeats = math.ceil(config.total_steps / number_of_batches)
-    print(f"Repeating dataset {num_repeats} times")
-
-    repeated_train_ds = input_utils.get_data(
-        dataset=train_subset_data_builder,
-        split=config.train_split,
-        rng=train_ds_rng,
-        process_batch_size=local_batch_size,
-        preprocess_fn=preprocess_spec.parse(
-            spec=config.pp_train, available_ops=preprocess_utils.all_ops()),
-        shuffle_buffer_size=config.shuffle_buffer_size,
-        prefetch_size=config.get("prefetch_to_host", 2),
-        # TODO(joost,andreas): double check if below leads to bootstrap sampling
-        num_epochs=num_repeats,
-    )
+    # Only fine-tune if there is anything to fine-tune with.
+    if current_train_ds_length > 0:
+      # Repeat dataset to have oversampled epochs and bootstrap more batches
+      number_of_batches = current_train_ds_length / config.batch_size
+      num_repeats = math.ceil(config.total_steps / number_of_batches)
+      print(f"Repeating dataset {num_repeats} times")
 
-    train_eval_ds = input_utils.get_data(
-        dataset=train_subset_data_builder,
-        split=config.train_split,
-        rng=train_ds_rng,
-        process_batch_size=local_batch_size,
-        preprocess_fn=preprocess_spec.parse(
-            spec=config.pp_eval, available_ops=preprocess_utils.all_ops()),
-        shuffle=False,
-        drop_remainder=False,
-        prefetch_size=config.get("prefetch_to_host", 2),
-        num_epochs=1,
-    )
+      repeated_train_ds = input_utils.get_data(
```

Comment:
what is `repeated_train_ds` and its difference to `train_eval_ds`? Please add some comments for context.

Reply:
See added comments in code.